dialogue-clients/src/main/java/com/palantir/dialogue/core/DialogueClients.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.google.errorprone.annotations.CheckReturnValue;
+import com.palantir.conjure.java.api.config.service.ServiceConfiguration;
+import com.palantir.conjure.java.api.config.service.ServicesConfigBlock;
+import com.palantir.conjure.java.api.config.service.UserAgent;
+import com.palantir.conjure.java.client.config.ClientConfiguration;
+import com.palantir.conjure.java.client.config.NodeSelectionStrategy;
+import com.palantir.dialogue.ConjureRuntime;
+import com.palantir.refreshable.Refreshable;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
+
+/**
+ * Only this class is public API, everything else is package private and may change.
+ *
+ * All of the interfaces have been split up into separate chunks so that libraries can depend on only as much as they
+ * need. This also makes it more convenient to implement these interfaces elsewhere (e.g. in WC).
+ */
+public final class DialogueClients {
+
+    public static Factory create() {
+        return DefaultFactory.create();
+    }
+
+    public interface ReloadingClients {
+        <T> T get(Class<T> serviceClass, String serviceName);
```

Comment:
Do we want to expose the `.class` reflection-based factory here? We could do that internally in wc, but take a function in this factory to minimize complexity.

Reply:
We could completely isolate channel creation from service creation, probably out of scope.